File: 2483_Minimum_Penalty_for_a_Shop/solution.c
int bestClosingTime(char* customers) {
    int penalty = 0, min_penalty = 0, best_hour = 0;
    for (int i = 0; customers[i] != '\0'; i++) if (customers[i] == 'Y') penalty++;

    min_penalty = penalty;
    for (int i = 0; customers[i] != '\0'; i++) {
        customers[i] == 'Y' ? penalty-- : penalty++;
        if (penalty < min_penalty) min_penalty = penalty, best_hour = i + 1;
    }
    return best_hour;
}

File: 2483_Minimum_Penalty_for_a_Shop/solution.py
class Solution:
    def bestClosingTime(self, customers: str) -> int:
        penalty = customers.count("Y")
        min_penalty, best_hour = penalty, 0

        for i in range(len(customers)):
            penalty += -1 if customers[i] == "Y" else 1
            if penalty < min_penalty:
                min_penalty = penalty
                best_hour = i + 1

        return best_hour
